### PR TITLE
feature(Tiktok): add support to take user properties from traits

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> add support to take user properties from `properties.context.user | traits | context.traits`

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/tiktok_ads/transform.js [hunk 1]
```diff
@@ -52,6 +53,20 @@ const trackResponseBuilder = async (message, { Config }) => {
...
+  let email = get(payload, "context.user.email");
+  if (!isDefinedAndNotNullAndNotEmpty(email) && traits?.email) {
+    set(payload, "context.user.email", traits.email);
+  }
+
+  let phone_number = get(payload, "context.user.phone_number");
+  if (!isDefinedAndNotNullAndNotEmpty(phone_number) && traits?.phone) {
+    set(payload, "context.user.phone_number", traits.phone);
+  }
+
```
Comment (ItsSudip) [inline v0/destinations/tiktok_ads/transform.js:69]: It is not the right way to collect user traits. We get user traits in context.traits after an identify call. We should map that to context.user in the resultant payload, not from properties.context.user.

Comment (akashrpo) [reply v0/destinations/tiktok_ads/transform.js:69]: We are taking traits from [`traits`, `context.traits`] L:57. and mapping that to `context.user.xxx` in resultant payload

Comment (akashrpo) [reply v0/destinations/tiktok_ads/transform.js:69]: We just have email and phone. Ref - https://ads.tiktok.com/gateway/docs/index?identify_key=2b9b4278e47b275f36e7c39a4af4ba067d088e031d5f5fe45d381559ac89ba48&language=ENGLISH&doc_id=1727541103358977#item-link-Context%20object%20parameters

Comment (ItsSudip) [reply v0/destinations/tiktok_ads/transform.js:69]: Accha got it.

**Review by ItsSudip — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**